stores/amazon_monitoring.py

```diff
@@ -287,15 +286,17 @@ async def stock_check(self, queue: asyncio.Queue, future: asyncio.Future):
             await wait_timer(end_time)
             end_time = time.time() + delay
             status, response_text = await self.aio_get(url=self.item.furl.url)
-            save_html_response("stock-check", status, response_text)
+            # save_html_response("stock-check", status, response_text)
+            collector.collect(status, self.connector)
             # do this after each request
             fail_counter = check_fail(status=status, fail_counter=fail_counter)
             if fail_counter == -1:
                 session = self.fail_recreate()
                 future.set_result(session)
                 return
 
-            check_count += 1
+            self.check_count += 1
+            self.next_item()
```

Comment:
This isn't going to work, I don't think, because it looks like the way you've implemented `next_item` it returns the next item, but doesn't update `self.item` to reference the next item. It's an easy fix to add that, though.

Reply:
as soon as the built-in next() method is used the cycle object is automatically updated though I'll look further into it.

Reply:
I now see what you actually mean. The `next_item()` method does set the `self.item` attribute of an `AmazonMonitor` object. I do see how that can be confusing. Might have to make it more clear.

Reply:
Oh no I really really see this time what I did wrong. What a shitshow. Good eye.